pkg/distributor/distributor.go

```diff
@@ -498,9 +586,28 @@ func (d *Distributor) sendSamples(ctx context.Context, ingester ring.IngesterDes
 	}
 	_, err = c.Push(ctx, &req)
 
-	ingesterAppends.WithLabelValues(ingester.Addr).Inc()
+	ingesterAppends.WithLabelValues(ingester.Addr, typeSamples).Inc()
+	if err != nil {
+		ingesterAppendFailures.WithLabelValues(ingester.Addr, typeSamples).Inc()
+	}
+	return err
+}
+
+func (d *Distributor) sendMetadata(ctx context.Context, ingester ring.IngesterDesc, metadata []*client.MetricMetadata) error {
```

Comment:
`sendSamples()` and `sendMetadata()` can be easily merged into a single function (ie. `sendToIngester()`) which takes in input the `client.WriteRequest{}` and `typeSamples/typeMetadata`. However, to keep this diff easier, I'm OK if you work on it on a separate PR (along with deduplicating the `ring.Do()` logic).

Reply:
Should have a PR right after 😄